hydra-node/test/Hydra/HeadLogicSpec.hs

```diff
@@ -239,6 +240,39 @@ spec = do
           let s' = update env ledger s rollback
           void $ run $ s' `hasEffect` ClientEffect RolledBack
 
+      it "contests when detecting close with old snapshot" $ do
+        let snapshot = Snapshot 2 mempty []
+            latestConfirmedSnapshot = ConfirmedSnapshot snapshot (aggregate [])
+            s0 =
+              inOpenState' threeParties $
+                CoordinatedHeadState
+                  { seenUTxO = mempty
+                  , seenTxs = mempty
+                  , confirmedSnapshot = latestConfirmedSnapshot
+                  , seenSnapshot = NoSeenSnapshot
+                  }
+            closeTxEvent = OnChainEvent $ Observation $ OnCloseTx 0 42
+            contestTxEffect = OnChainEffect $ ContestTx latestConfirmedSnapshot
+        s1 <- update env ledger s0 closeTxEvent `hasEffect` contestTxEffect
+        s1 `shouldSatisfy` \case
+          ClosedState{} -> True
+          _ -> False
+
+      it "re-contests when detecting contest with old snapshot" $ do
+        let snapshot2 = Snapshot 2 mempty []
+            latestConfirmedSnapshot = ConfirmedSnapshot snapshot2 (aggregate [])
+            s0 = inClosedState' threeParties latestConfirmedSnapshot
+            contestSnapshot1Event = OnChainEvent $ Observation $ OnContestTx 1
+            contestTxEffect = OnChainEffect $ ContestTx latestConfirmedSnapshot
+        s1 <- update env ledger s0 contestSnapshot1Event `hasEffect` contestTxEffect
+        s1 `shouldSatisfy` \case
+          ClosedState{} -> True
+          _ -> False
+
+      -- TODO: Maybe we want to instead ensure that the headlogic contests until
+      -- it's latest known snapshot is closed
+      it "re-contests if contest transaction fails" pending
```

Comment:
Ah maybe we should remove this test. After our discussion I think it's not the responsibility of the `hydra-node` to resubmit failed transactions. We might want to provide a way to contest again to our clients.. sounds oddly familiar.

Reply:
Aaand.. there is even a `ClientInput` named [`Contest`](https://github.com/input-output-hk/hydra-poc/blob/d38177553dc6f26e69685052587d5013fb0a4b5e/hydra-node/src/Hydra/ClientInput.hs#L17), which is unhandled in the HeadLogic right now! What would be the semantics of that? Let's discuss this tomorrow.